Lib/asyncio/selector_events.py

```diff
@@ -589,7 +589,10 @@ class _SelectorTransport(transports._FlowControlMixin,
     def __init__(self, loop, sock, protocol, extra=None, server=None):
         super().__init__(extra, loop)
         self._extra['socket'] = sock
-        self._extra['sockname'] = sock.getsockname()
+        try:
+            self._extra['sockname'] = sock.getsockname()
+        except socket.error:
```

Comment:
I believe the proper exception type is `OSError`.

Reply:
Indeed!